src/item-navigator.ts

```diff
@@ -0,0 +1,532 @@
+import {
+  css,
+  html,
+  LitElement,
+  customElement,
+  property,
+  state,
+  query,
+  PropertyValues,
+  CSSResult,
+} from 'lit-element';
+import { nothing, TemplateResult } from 'lit-html';
+import { MetadataResponse } from '@internetarchive/search-service';
+import {
+  SharedResizeObserver,
+  SharedResizeObserverResizeHandlerInterface,
+} from '@internetarchive/shared-resize-observer';
+// @ts-ignore
+import { IAMenuSlider } from '@internetarchive/ia-menu-slider';
+import '@internetarchive/ia-menu-slider';
+
+import { ModalManagerInterface } from '@internetarchive/modal-manager';
+import '@internetarchive/icon-ellipses/icon-ellipses';
+import './loader';
+
+import {
+  IntManageSideMenuEvent,
+  IntSetOpenMenuEvent,
+  IntSetMenuContentsEvent,
+  IntSetMenuShortcutsEvent,
+  IntLoadingStateUpdatedEvent,
+  IntManageFullscreenEvent,
+} from './interfaces/event-interfaces';
+
+import { IntMenuProvider, IntMenuShortcut } from './interfaces/menu-interfaces';
+import './no-theater-available';
+@customElement('ia-item-navigator')
+export class ItemNavigator
+  extends LitElement
+  implements SharedResizeObserverResizeHandlerInterface {
+  @property({
+    type: Object,
+    converter: (value: string | MetadataResponse | null): MetadataResponse => {
+      if (value && typeof value === 'string') {
+        return new MetadataResponse(JSON.parse(atob(value)));
+      }
+      return value as MetadataResponse;
+    },
+  })
+  item: MetadataResponse | undefined = undefined;
+
+  @property({ type: String }) itemType = '';
+
+  @property({ type: String }) baseHost = 'archive.org';
+
+  @property({
+    converter: (arg: string | boolean | null) => {
+      if (typeof arg === 'boolean') {
+        return arg;
+      }
+      return arg === 'true';
+    },
+  })
+  signedIn = false;
+
+  @property({
+    type: Array,
+    hasChanged: (newVal: Array<object>, oldVal: Array<object>) => {
+      if (newVal !== oldVal) {
+        return true;
+      }
+      return false;
+    },
+  })
+  @property({ type: Array })
+  menuContents: IntMenuProvider[] = [];
+
+  @property({ type: Array }) menuShortcuts: IntMenuShortcut[] = [];
+
+  @property({ type: Boolean, reflect: true }) viewportInFullscreen:
+    | boolean
+    | null = null;
+
+  @property({ type: Boolean }) menuOpened = false;
+
+  @property({ type: String }) openMenu = '';
```

Comment:
Can the `openMenu` be enumerated so we don't pass in an arbitrary string into this?

Reply:
thanks for catching. this string ties in with a shortcut & a side panel. i've created `MenuId` to thread through.